mlflow/types/schema.py

```diff
@@ -122,34 +125,280 @@ def from_numpy_type(cls, np_type):
         return next((v for v in cls._member_map_.values() if v.to_numpy() == np_type), None)
 
 
+ArrayType = TypeVar("ArrayType", bound="Array")
+ObjectType = TypeVar("ObjectType", bound="Object")
```

Comment:
Do we need these? Can we directly use `Array` or `Object`?

Reply:
We can't. Property and Array, Object depend on each other. By the time we create Property, Array or Object is not defined yet.

Reply:
can we use forward references?

Reply:
https://stackoverflow.com/questions/55320236/does-python-evaluate-type-hinting-of-a-forward-reference